queries/appSettings.ts

```diff
@@ -1,81 +1,61 @@
 import { createId } from '@paralleldrive/cuid2';
 import { redirect } from 'next/navigation';
 import 'server-only';
+import { type z } from 'zod';
+import { createAppSetting } from '~/actions/appSettings';
 import { UNCONFIGURED_TIMEOUT } from '~/fresco.config';
 import { createCachedFunction } from '~/lib/cache';
+import { type appSettingSchema } from '~/schemas/appSettings';
 import { prisma } from '~/utils/db';
 
-type AppSettingsKeys =
-  | 'configured'
-  | 'allowAnonymousRecruitment'
-  | 'limitInterviews'
-  | 'initializedAt'
-  | 'installationId';
-
-type AppSettings = Record<AppSettingsKeys, string>;
-
-const DEFAULT_SETTINGS: Record<AppSettingsKeys, string> = {
-  configured: 'false',
-  allowAnonymousRecruitment: 'false',
-  limitInterviews: 'true',
-  initializedAt: new Date().toISOString(),
-  installationId: createId(),
-};
+// Generic function to get an app setting
+export async function getAppSetting<
+  Key extends keyof z.infer<typeof appSettingSchema>,
+>(key: Key): Promise<z.infer<typeof appSettingSchema>[Key]> {
+  const initializedAt = await prisma.appSettings.findUnique({
+    where: { key: 'initializedAt' },
+  });
 
-const calculateIsExpired = (configured: boolean, initializedAt: Date) =>
-  !configured && initializedAt.getTime() < Date.now() - UNCONFIGURED_TIMEOUT;
+  if (!initializedAt) {
+    // initialize the app
+    // todo: should these come from fresco.config?
```

Comment:
Yes, and then I would make a separate function for `initialiseWithDefaults` or something.

Reply:
You also should think about putting this in a transaction to ensure that they all succeed, or are rolled back. 

If you can't work out how to do that using `createAppSetting`, I think its okay to have the `initialiseWithDefaults` bulk create records with a one-off query.

Reply:
created initializeWithDefaults as a one-off query in a transaction [b9da84f](https://github.com/complexdatacollective/Fresco/pull/161/commits/b9da84f06fc178d6c9906a2aabf7e3958146c5d7). Note that I had to increase the timeout from the default bc this transaction takes longer than the default 5 seconds. Need to think about if this will cause problems with the vercel free tier like we've run into in the past.